subsys/net/ip/dhcpv4.c

```diff
@@ -150,15 +150,22 @@ static inline bool dhcpv4_add_sname(struct net_pkt *pkt)
 /* Create DHCPv4 message and add options as per message type */
 static struct net_pkt *dhcpv4_create_message(struct net_if *iface, u8_t type,
 					     const struct in_addr *ciaddr,
+					     const struct in_addr *src,
 					     const struct in_addr *server_addr,
 					     bool server_id, bool requested_ip)
 {
 	NET_PKT_DATA_ACCESS_DEFINE(dhcp_access, struct dhcp_msg);
-	const struct in_addr src = INADDR_ANY_INIT;
+	const struct in_addr any = INADDR_ANY_INIT;
```

Comment:
This "any" variable can be removed as there exists a function `net_ipv4_unspecified_address()` that will return the any address to the caller.

Reply:
Done, that's a better one to use.